iputil/ip.go

```diff
@@ -0,0 +1,70 @@
+// package iputil defines type that will marshal/unmarshal an IP address as
+// human readable IPv4 string rather than an unreadable byte stream.
+package iputil
+
+import (
+	"bytes"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"net"
+)
+
+// IP is a special wrapped type of net.IP
+type IP net.IP
+
+// Gt will return if the IP is greater than the given IP
+func (ip IP) Gt(start net.IP) bool {
+	return bytes.Compare(ip, start) >= 0
+}
+
+// Lt will return if the IP is less than the given IP
+func (ip IP) Lt(end net.IP) bool {
+	return bytes.Compare(ip, end) <= 0
+}
+
+// InRange will return if the IP is within the given range of addresses
+func (ip IP) InRange(start, end net.IP) bool {
+	return ip.Gt(start) && ip.Lt(end)
+}
+
+// String returns the IP as a string
+func (ip IP) String() string {
+	return net.IP(ip).String()
+}
+
+// Value will get the value for storing in SQL
+func (ip IP) Value() (driver.Value, error) {
+	return ip.String(), nil
+}
+
+// Scan will return the IP from the database
+func (ip *IP) Scan(value interface{}) error {
+	str, ok := value.([]byte)
+	if !ok {
+		return errors.New("invalid type stored in sql for whitelist ip")
```

Comment:
bit specific

Reply:
What would you suggest then mate

Reply:
```suggestion
		return fmt.Errorf("iputil: cannot scan type %T into ip.IP: %v", value, value)
```
seems a common format for an unsupported scanning type error